p2p/sentry/sentry_grpc_server.go

```diff
@@ -997,44 +998,60 @@ func (ss *GrpcServer) HandShake(context.Context, *emptypb.Empty) (*proto_sentry.
 	return reply, nil
 }
 
+func (ss *GrpcServer) startP2PServer(genesisHash libcommon.Hash) (*p2p.Server, error) {
+	if !ss.p2p.NoDiscovery {
+		if len(ss.discoveryDNS) == 0 {
+			if url := params.KnownDNSNetwork(genesisHash, "all"); url != "" {
+				ss.discoveryDNS = []string{url}
+			}
+		}
+		for _, p := range ss.Protocols {
+			dialCandidates, err := setupDiscovery(ss.discoveryDNS)
+			if err != nil {
+				return nil, err
+			}
+			p.DialCandidates = dialCandidates
+		}
+	}
+
+	srv, err := makeP2PServer(*ss.p2p, genesisHash, ss.Protocols)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = srv.Start(ss.ctx, ss.logger); err != nil {
+		srv.Stop()
```

Comment:
my opinion - it must be responsibility of func `Start` (there must be defer which will call Stop - if err/panic happened). but let's keep it out of this PR scope.